pkg/dependencies/factory.go

```diff
@@ -190,23 +193,51 @@ func (f *Factory) WithExecutableMountDirs(mountDirs ...string) *Factory {
 	return f
 }
 
+func (f *Factory) WithLocalExecutables() *Factory {
+	f.executablesInDocker = false
+	return f
+}
+
+// UserExecutablesDockerClient forces a specific DockerClient to build
+// Executables as opposed to follow the normal building flow
+// This is only for testing
+func (f *Factory) UserExecutablesDockerClient(client executables.DockerClient) *Factory {
```

Comment:
Is it supposed to be this?
```suggestion
func (f *Factory) UseExecutablesDockerClient(client executables.DockerClient) *Factory {
```

Reply:
If this is only used for testing, it might be clearer to mention in the method name that it is for Testing. But I would understand not doing that if we want to extend it to be able to pass it in other situations